server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1059,6 +1077,42 @@ public void onFailure(Exception ex) {
         }
     }
 
+    private void releaseResourceLockFiles (
+        Collection<SnapshotId> snapshotIds,
+        RepositoryData repositoryData,
+        RemoteStoreLockManagerFactory remoteStoreLockManagerFactory,
+        ActionListener<Void> onAllShardsCompleted
+    ){
+        for (SnapshotId snapshotId: snapshotIds) {
+            List<String> indices = this.getSnapshotInfo(snapshotId).indices();
+            List<IndexId> indexIds = repositoryData.resolveIndices(indices);
+            for (IndexId indexId: indexIds) {
+                try {
+                    IndexMetadata indexMetadata = this.getSnapshotIndexMetaData(repositoryData,
+                        snapshotId, indexId);
+                    if (this.getSnapshotInfo(snapshotId).isRemoteStoreInteropEnabled()
+                        && indexMetadata.getSettings().getAsBoolean(
+                        IndexMetadata.SETTING_REMOTE_STORE_ENABLED,
+                        false)) {
```

Comment:
I think the second check should be an assertion instead. I don't see a case where `SnapshotInfo` tells the snapshot is with shallow copy and indexSettings doesn't have Remote store enabled. Since shallow copy case only works on the remote store backed indices.